docs/messaging/messaging-spans.md

```diff
@@ -269,6 +271,12 @@ batch of messages, or for no message at all (if it is signalled that no
 messages were received). For each message it accounts for, the "Deliver" or
 "Receive" span SHOULD link to the message's creation context.
 
+"Settle" spans SHOULD be created for every manually or automatically triggered
+settlement operation. A single "Settle" span can account for a single message
+or for multiple messages (in case messages are passed for settling as batches).
+For each message it accounts for, the "Settle" span MAY link to the creation
```

Comment:
not blocking: why MAY and not SHOULD? (which implies 'if possible')

Reply:
I took that over from the OTEP.

The MAY gives people more possibilities to model their scenarios. For example [this proposal for GCP Pub/Sub](https://github.com/open-telemetry/semantic-conventions/blob/fb3723986ec78e65e1d6a00eec483ed0db0a9f0f/docs/messaging/gcp-pubsub.md#unary-pull-example) (which still might need some discussion) links settlement-like operations (acks) to the "receive" span. That might be more intuitive in some scenarios?

Reply:
If I recall we discussed the proposal with @alevenberg and came to the conclusion that link to receive span is MAY, but link to the message is SHOULD. Similar to how we do deliver ---link--> message, but also deliver MAY be a child of message.

This way we have certainty on the relationship - settle is linked to the message (when possible) and flexibility - there could be other links if that's beneficial. 

The assumption is that by default tracing message e2e is important and knowing when/how message was settled is a higher priority than receive/rpc call that received it.

If settle is linked to to the message, it puts it 1 link away. Linking it to receive puts settle 2 links away from the message complicating the queries and joins, and requiring even more luck with sampling (to collect everything).

Reply:
As said above, I kept the MAY because it is in the OTEP, it supports @alevenberg's proposal above, and it maybe also leaves some more room for experimentation during prototyping.

However, I'll bring it up in the next meeting, I'd be fine making it a SHOULD if there's consensus around that. Also, this is something that will be exempt from initial stability, so we can still revisit this after stabilizing conventions.

Reply:
sure, let's not block on this

Reply:
Not blocking: but the reasoning for linking to the receive span instead of the create span is if there are multiple subscription resources.

For example, if a message is published to topic-a and then received on sub-a and sub-b, you would want the settle span to link to the corresponding subscription span directly.

Reply:
I think we would achieve a similar effect by setting subscription name attribute on a settlement span. 

without deliver or process span user experience would not be great either way and with it links wouldn't matter that much. So it's definitely not a hill I want to die on :)

Reply:
Yeah I think the issue was if it is a parent/child relationship between the create and receive and it had multiple subs if we link to the create, it's not as clear what sub it was a part of. 

I think keeping it as MAY is nice to give the user the flexibility depending on their instrumentation.